scripts/check-mcp-compliance.py

```
@@ -0,0 +1,262 @@
+            for py_file in dir_path.rglob('*.py'):
+                if 'test' in py_file.name or '__pycache__' in str(py_file):
+                    continue
+                    
+                try:
+                    content = py_file.read_text()
+                    for pattern in mock_patterns:
+                        if pattern in content:
+                            # Count occurrences
+                            count = content.count(pattern)
+                            issues.append(f"{py_file}: {count} occurrences of '{pattern}'")
+                except Exception as e:
```

Ruff (0.11.9)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)